### ISSUE TITLE
Is it possible to use Steam Direct Mode implement ATW(ASW)?

### ISSUE THREAD

**Issue by vicdxxx** [ValveSoftware/openvr#615]
I am done openvr direct mode on my hmd, but I want to achieve time warp in my driver. 
Did anyone know is that possible?
Is there some limit in direct mode? I feel direct mode is still in beta, because some game summit layers show juddery, jerky.
Any suggestion is highly appreciate!

**Comment by JoeLudwig** [comment]
If at all possible you should let the SteamVR compositor talk to the display directly and not re-implement your own direct mode. You just need to return the VID and PID from the EDID and we can handle the rest. That will get you timewarp and all the other compositor features without any extra work.

Are you providing velocity and angular velocity with your poses? If not, fix that first because that will definitely cause thing to be jittery.

**Comment by vicdxxx** [comment]
@JoeLudwig Is that mean I should use the steam direct mode, and register my VID, PID rather than  use my own direct mode ? And use their reprojection setting in steamvr to get ASW effects?
I did not provide angular velocity data. thank you so much! However, there is not obviously jittery somehow. But I will add it as you mentioned.

**Comment by JoeLudwig** [comment]
Yes, that should be significantly easier than implementing driver direct mode and reprojection on your own.

**Comment by TheDeveloperGuy** [comment]
@JoeLudwig Will it ever end, this confusion between Oculus-Direct and VR-Direct? 😄

**Comment by Balderick** [comment]
I do not think it is different vr platforms supporting direct mode as causing the confusion but proprietary GPU drivers and the white listing process as cause.

**Comment by TheDeveloperGuy** [comment]
@Balderick There are 2 completely different concepts at play here which are both called "direct" mode but are not the same thing at all. I'm not going to bother explaining it again because I already have done that lots of times here. This topic (and your comment) highlight this misunderstanding. If you want to know the difference look through my posts here and you will find several of the times I've explained it. ;)

**Comment by Balderick** [comment]
Sharing a link to the info you refer too might be helpful.

**Comment by TheDeveloperGuy** [comment]
One of us has to search. It may as well being the one seeking the info, rather than the one that's done their best to explain it to a deaf audience... :wink:

**Comment by Balderick** [comment]
Well vicdxxx and I make two people .    You know better than Valve developer?

**Comment by TheDeveloperGuy** [comment]
@Balderick Cool story Bro. It was a comment to @JoeLudwig not anyone else.

**Comment by Balderick** [comment]
Ehm private messages are private.   Posts in public forums are public.

**Comment by TheDeveloperGuy** [comment]
@vicdxxx Anyway, the takeaway from this is that you shouldn't be using IVRDriverDirectModeComponent for your driver unless you really know what you are doing. You should be using IVRDisplayComponent in most cases that involve a HMD that is connected via HDMI.

**Comment by TheDeveloperGuy** [comment]
@Balderick Looks like you're a person that needs to be given fish, not taught how to fish, so here's your fish.... :wink:

https://github.com/ValveSoftware/openvr/issues/431#issuecomment-284415766
